### ISSUE TITLE
Creation of a new type of view: Swipe

### ISSUE THREAD

**Issue by thomaspiller** [unep-grid/map-x-mgl#591]
This new type of view should offer the same options as what has been developed for dashboards (see [here](https://app.mapx.org/static.html?views=MX-DBG8E-HEPLU-6YV22&zoomToViews=true&language=en&)).

The basic specs are:
- swipe between 2 views displayed in the map
- more than 2 views can be compared within the same swipe view -> presence of drop-down lists to choose which views are displayed on each side of the swipe
- views usable in the swipe must be published in the current project and the swipe editor must have read rights on them
- view features such as filters, popup, time-slider, legend should be available in swipe views
- reading and editing rights for swipe views are managed as for other types of views

**Comment by antobenve** [comment]
@thomaspiller can you clarify if the views accepted in the swipe view can also be those that are shared into the project or only those that are strictly "published" into the project? I think we should allow flexibility if possible.

**Comment by thomaspiller** [comment]
I agree, views that are shared to the project should be usable in a swipe view.

**Comment by fxi** [comment]
Other solutions to avoid dropdowns  AND duplicating all views filters, buttons, options, legend and al.

**A new button "Toggle swipe mode for this view". The view is displayed in the right part of the screen.**
![image](https://user-images.githubusercontent.com/1196833/96740116-ceda6580-13c0-11eb-9d08-ae16b8587a71.png)

<hr>

**Drag and drop view to the right of the screen, the view open normally with all option, the layers are displayed on a swipe panel.**
<img width="1052" alt="Screenshot 2020-10-21 at 17 45 42" src="https://user-images.githubusercontent.com/1196833/96744626-8a9d9400-13c5-11eb-9a15-c6cd62d67671.png">

For me, the condition to have a new view of type "Swipe" is that only two views are selected in a configuration panel by a publisher : if it's more complexe than that, it will be hell to use them in story maps, static mode and SDK.

**Comment by antobenve** [comment]
Ok, based on your proposition will the swipe mode be available in story maps then?
Furthermore, I like the idea to propose a different approach but we must make sure it fullfills the requirements of contributors too. In this case the requirement is to have products that offer the comparison between selected data. Another requirement is to be able to set more than 1 data per screenside (the need of dropdowns). Therefore, the product must be easy to find and to use and to show direct comparison between specific maps on which the user has some (limited) action.
How can we follow your flexible approach while being able to fullfill the requirements?  
Would the SDK an elegant solution to reach the requirements? I think so if the partner has a webpage where it sets the views to compare using dropdown lists while the swipe mode is toggled on on map by default.
